lightning/src/chain/keysinterface.rs

```diff
@@ -545,40 +676,40 @@ impl KeysManager {
 					channel_id_master_key,
 					channel_id_child_index: AtomicUsize::new(0),
 
-					unique_start,
+					seed: *seed,
+					starting_time_secs,
+					starting_time_nanos,
 				}
 			},
 			Err(_) => panic!("Your rng is busted"),
 		}
 	}
-}
-
-impl KeysInterface for KeysManager {
-	type ChanKeySigner = InMemoryChannelKeys;
-
-	fn get_node_secret(&self) -> SecretKey {
-		self.node_secret.clone()
-	}
-
-	fn get_destination_script(&self) -> Script {
-		self.destination_script.clone()
-	}
-
-	fn get_shutdown_pubkey(&self) -> PublicKey {
-		self.shutdown_pubkey.clone()
+	fn derive_unique_start(&self) -> Sha256State {
+		let mut unique_start = Sha256::engine();
+		unique_start.input(&byte_utils::be64_to_array(self.starting_time_secs));
+		unique_start.input(&byte_utils::be32_to_array(self.starting_time_nanos));
+		unique_start.input(&self.seed);
+		unique_start
 	}
+	/// Derive an old set of ChannelKeys for per-channel secrets based on a key derivation
+	/// parameters.
+	/// Key derivation parameters are accessible through a per-channel secrets
+	/// ChannelKeys::key_derivation_params and is provided inside DynamicOuputP2WSH in case of
+	/// onchain output detection for which a corresponding delayed_payment_key must be derived.
+	pub fn derive_channel_keys(&self, channel_value_satoshis: u64, params_1: u64, params_2: u64) -> InMemoryChannelKeys {
+		let chan_id = ((params_1 & 0xFFFF0000) >> 32) as u32;
```

Comment:
Why is this getting the middle 2 bytes instead of the top 4 bytes? It'd be nice if the test were sufficiently strict to catch this.

Reply:
Oh, duh, this isnt a bug, the shift effectively clears those bits anyway, you can drop the "&".

Reply:
That's a good point, we are dealing with u64 here, my bad. Hardened test to detect key set collision.